stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -446,16 +452,46 @@ void swift::removeStatusRecord(AsyncTask *task, TaskStatusRecord *record,
 
 }
 
-// Convenience wrapper for when we don't care to make any modifications to the
-// flags fields of the active task status when we are removing a task status
-// record
+// Convenience wrapper for when client hasnt already done the load of the status
 SWIFT_CC(swift)
-void swift::removeStatusRecord(TaskStatusRecord *record) {
+void swift::removeStatusRecord(AsyncTask *task, TaskStatusRecord *record,
+     llvm::function_ref<void(ActiveTaskStatus, ActiveTaskStatus&)>fn) {
+  auto oldStatus = task->_private()._status().load(std::memory_order_relaxed);
+  return removeStatusRecord(task, record, oldStatus, fn);
+}
 
-  auto task = swift_task_getCurrent();
-  SWIFT_TASK_DEBUG_LOG("remove status record = %p, from current task = %p",
-                       record, task);
-  return removeStatusRecord(task, record);
+// Convenience wrapper for modifications on current task
+SWIFT_CC(swift)
+void swift::removeStatusRecordFromSelf(TaskStatusRecord *record, ActiveTaskStatus &status,
+     llvm::function_ref<void(ActiveTaskStatus, ActiveTaskStatus&)>fn) {
+  return removeStatusRecord(swift_task_getCurrent(), record, status, fn);
+}
+
+SWIFT_CC(swift)
+void swift::removeStatusRecordFromSelf(TaskStatusRecord *record,
+     llvm::function_ref<void(ActiveTaskStatus, ActiveTaskStatus&)>fn) {
+  return removeStatusRecord(swift_task_getCurrent(), record, fn);
+}
+
+SWIFT_CC(swift)
+void swift::updateStatusRecord(AsyncTask *task, TaskStatusRecord *record,
+     llvm::function_ref<void()>updateRecord,
+     ActiveTaskStatus& status,
+     llvm::function_ref<void(ActiveTaskStatus, ActiveTaskStatus&)>fn) {
+
+  withStatusRecordLock(task, status, [&](ActiveTaskStatus lockedStatus) {
+#if NDEBUG
```

Comment:
Should be `#ifndef NDEBUG`.

Reply:
Oh yes. Okay will do this in follow up PR.